README.md

```diff
@@ -9,67 +8,67 @@ Lead Maintainer - [Devin Ivy](https://github.com/devinivy)
+Underdog brings [HTTP/2 server-push](http://httpwg.org/specs/rfc7540.html#PushResources) to **hapi v17+**.  The way it works is that you specify paths to resources that you'd like to push alongside a particular response.  This is achieved with a call to the response toolkit decoration [`h.push()`](API.md#hpushresponse-path-headers).  Before hapi responds to the original request, those push-requests will be made internally and their results will be streamed to the client as push-responses.  Even pushed resources can specify additional resources to push.  You can't make this stuff up!
 
 ### Example
 ```js
 const Fs = require('fs');
+const Http2 = require('http2');
 const Hapi = require('hapi');
-const Http2 = require('http2'); // Or use node-spdy with h2
 const Underdog = require('underdog');
 
-const listener = Http2.createServer({
-    // See tests for a key/cert that you can use to try this out
-    key: Fs.readFileSync(`${__dirname}/localhost.key`),
-    cert: Fs.readFileSync(`${__dirname}/localhost.cert`)
-});
+(async () => {
 
-const server = new Hapi.Server();
-server.connection({ listener, port: 3000 });
+    const listener = Http2.createSecureServer({
+        // See tests for a key/cert that you can use to try this out
+        key: Fs.readFileSync(`${__dirname}/localhost.key`),
+        cert: Fs.readFileSync(`${__dirname}/localhost.cert`)
+    });
 
-server.register(Underdog, (err) => {
+    const server = Hapi.server({
+        listener,
+        tls: true,
+        port: 3000
+    });
 
-    if (err) {
-        throw err;
-    }
+    await server.register(Underdog);
 
     server.route([
         {
             method: 'get',
             path: '/',
-            handler: (request, reply) => {
+            handler: (request, h) => {
+
+                const response = h.response('<script src="/push-me.js"></script>');
 
-                const response = reply('<script src="/push-me.js"></script>');
+                h.push(response, 'push-me.js');
 
-                reply.push(response, 'push-me.js');
+                return response;
             }
         },
         {
             method: 'get',
             path: '/push-me.js',
-            handler: (request, reply) => {
+            handler: (request) => {
 
-                reply('document.write(\'I was definitely pushed!\');');                
+                return 'document.write(\'I was definitely pushed!\');';                
             },
             // To demonstrate that it must have been pushed, not requested directly
             config: { isInternal: true }
         }
     ]);
 
-    server.start((err) => {
+    await server.start();
 
-        if (err) {
-            throw err;
-        }
-
-        console.log(`Check-out ${server.info.uri} in your favorite HTTP/2-supporting client`);
-    });
-});
+    console.log(`Check-out ${server.info.uri} in your favorite HTTP/2-supporting client`);
+})();
```

Comment:
yo! my bad I'm late here. 

this example is pretty clear, but one thing I was semi-looking for (and got by just reading the damn code, so prolly ignore me ;) ) is a concise description of what's expected to happen; i.e. , essentially, however you might describe this test case in the test file. 

again, I got it from both your introduction and reading the code, all pretty clear even to me, but I guess feeling an _even shorter_ description would help learn this a bit quicker. small, small potatoes, take or leave.

this is such a cool tool; i tried to process the rest of the work, my brain broke a bit doing so, but I'd like to make time to come back and actually process what's happening here. Thanks for writing this up and sharing 🙏